multi/stm32l4-multi/tty.c

```diff
@@ -63,9 +69,31 @@ typedef struct {
 
 	libtty_common_t tty_common;
 
-	uint8_t rx_fifo_buffer[16];
-	lf_fifo_t rx_fifo;
-	volatile int rxready;
+	enum {
+		tty_irq,
+		tty_dma,
+	} type;
+	union {
+		struct {
+			uint8_t rx_fifo_buffer[16];
```

Comment:
Buffer allocation and size configuration could be used for both irq and dma handlers.

Reply:
added comment.

Reply:
I think it should be added along side the option to mix dma and irq usage on one uart in a separate PR once needed